pkg/client/build.go

```diff
@@ -939,7 +953,34 @@ func prependBuildpackToOrder(order dist.Order, bpInfo dist.ModuleInfo) (newOrder
 	return newOrder
 }
 
-func (c *Client) createEphemeralBuilder(rawBuilderImage imgutil.Image, env map[string]string, order dist.Order, buildpacks []buildpack.BuildModule) (*builder.Builder, error) {
+func (c *Client) processExtensions(ctx context.Context, builderImage imgutil.Image, builderExs []dist.ModuleInfo, builderOrder dist.Order, stackID string, opts BuildOptions) (fetchedExs []buildpack.BuildModule, orderExtensions dist.Order, err error) {
+	relativeBaseDir := opts.RelativeBaseDir
+	declaredExs := opts.Extensions
+
+	orderExtensions = dist.Order{{Group: []dist.ModuleRef{}}}
+	for _, ex := range declaredExs {
+		locatorType, err := buildpack.GetLocatorType(ex, relativeBaseDir, builderExs)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		switch locatorType {
+		case buildpack.RegistryLocator:
```

Comment:
A note, in case it helps others, `buildpack.FromBuilderLocator` is deprecated now that we have pre- and post- buildpacks. We may want `--pre-extension` and `--post-extension` (like exist for buildpacks) but that should be considered out-of-scope for this issue.

Reply:
I have added `case buildpack.FromBuilderLocator` in processExtensions()` funtion. Please help me review, thanks!